cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -154,300 +130,3 @@ def store_flow_cell_data(self, parsed_flow_cell: FlowCellDirectoryData) -> None:
-            flow_cell_id,
-            project_name,
-        )
-        return samples
-
-    @staticmethod
-    def sample_to_report_line(stats_sample: StatsSample, flow_cell_id: str) -> str:
-        lanes: str = ",".join(str(lane) for lane in stats_sample.lanes)
-        return "\t".join(
-            [
-                stats_sample.sample_name,
-                flow_cell_id,
-                lanes,
-                ",".join(str(unaligned.read_count) for unaligned in stats_sample.unaligned),
-                str(stats_sample.read_count_sum),
-                ",".join(str(unaligned.yield_mb) for unaligned in stats_sample.unaligned),
-                str(stats_sample.sum_yield),
-                ",".join(str(unaligned.q30_bases_pct) for unaligned in stats_sample.unaligned),
-                ",".join(str(unaligned.mean_quality_score) for unaligned in stats_sample.unaligned),
-            ]
-        )
-
-    @staticmethod
-    def get_report_lines(stats_samples: List[StatsSample], flow_cell_id: str) -> Iterable[str]:
-        """Convert stats samples to format lines ready to print."""
-        for stats_sample in sorted(stats_samples, key=lambda x: x.sample_name):
-            yield DemuxPostProcessingNovaseqAPI.sample_to_report_line(
-                stats_sample=stats_sample, flow_cell_id=flow_cell_id
-            )
-
-    @staticmethod
-    def write_report(report_path: Path, report_data: List[str]) -> None:
-        LOG.info("Write demux report to %s", report_path)
-        with report_path.open("w") as report_file:
-            report_file.write("\t".join(STATS_HEADER) + "\n")
-            for line in report_data:
-                report_file.write(line + "\n")
-
-    def get_report_data(self, flow_cell_id: str, project_name: Optional[str] = None) -> List[str]:
-        """Fetch the lines that are used to make a report from a flow cell."""
-        LOG.info(f"Fetch report data for flow cell: {flow_cell_id} project: {project_name}")
-        project_samples: List[StatsSample] = self.fetch_report_samples(
-            flow_cell_id=flow_cell_id, project_name=project_name
-        )
-
-        return list(self.get_report_lines(stats_samples=project_samples, flow_cell_id=flow_cell_id))
-
-    def create_cgstats_reports(self, demux_results: DemuxResults) -> None:
-        """Create a report for every project that was demultiplexed."""
-        flow_cell_id: str = demux_results.flow_cell.id
-        for project in demux_results.projects:
-            project_name: str = project.split("_")[-1]
-            report_data: List[str] = self.get_report_data(
-                flow_cell_id=flow_cell_id, project_name=project_name
-            )
-            report_path: Path = demux_results.demux_dir / f"stats-{project_name}-{flow_cell_id}.txt"
-            self.write_report(report_path=report_path, report_data=report_data)
-
-    @staticmethod
-    def create_barcode_summary_report(demux_results: DemuxResults) -> None:
```

Comment:
Do we need the barcode summary report for something? If so we need to implement this functionality.